src/Windows/Avalonia.Win32/Win32Platform.cs

```diff
@@ -137,13 +137,18 @@ public void ProcessMessage()
 
         public void RunLoop(CancellationToken cancellationToken)
         {
-            while (!cancellationToken.IsCancellationRequested)
+            var result = 0;
+            while (!cancellationToken.IsCancellationRequested 
+                && (result = UnmanagedMethods.GetMessage(out var msg, IntPtr.Zero, 0, 0)) > 0)
             {
-                UnmanagedMethods.MSG msg;
-                UnmanagedMethods.GetMessage(out msg, IntPtr.Zero, 0, 0);
                 UnmanagedMethods.TranslateMessage(ref msg);
                 UnmanagedMethods.DispatchMessage(ref msg);
             }
+            if (result < 0)
```

Comment:
Should we add a similar check here?

https://github.com/AvaloniaUI/Avalonia/blob/0f61b3cdd6d32f131041c68721f6216770378b9e/src/Windows/Avalonia.Win32/Win32Platform.cs#L133